utils/handles/sig.c

```diff
@@ -0,0 +1,22 @@
+#include "sig.h"
+
+#if !defined(_WIN32) && defined(__unix__) || defined(__unix) || \
+    (defined(__APPLE__) && defined(__MACH__))
+
+void signal_handler_posix(int signo) {
+  if (signo == ctrl_c_signal) {
+    signal_handler_core_end(signo);
```

Comment:
This API shouldn't be linked in any way to cIRI functions, it should be usable in any case.